### DISCUSSION TITLE
Status Badge

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#290] [Component suggestions]
<img width="480" alt="verschillende vormen van een status badge component" src="https://github.com/user-attachments/assets/46b74c55-1123-419b-b418-7b073957ada0">

We willen graag meer leren over de **Status Badge** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1733400002881449)

---

## Naam
Status Badge

## Doel
Label dat informeert over een status en de aandacht trekt met een signaalkleur.

## Ook bekend als
Badge, Tag

## Gerelateerde componenten
[Data Badge](https://github.com/orgs/nl-design-system/discussions/307), [Dot Badge](https://github.com/orgs/nl-design-system/discussions/308), [Number Badge](https://github.com/orgs/nl-design-system/discussions/214)

## Onderzoek
[Rapport Generieke Services 2023 - Gemeente Den Haag (PDF, 9 MB)](https://github.com/user-attachments/files/16993296/Rapport.generieke.services.23032023.def.pdf)
Uit dit gebruikersonderzoek blijkt dat de Status Badge heel goed werkt. Het wordt makkelijker begrepen dan het gebruik van enkel iconen.

[Onderzoek Generieke Services - Miro](https://miro.com/app/board/uXjVP0OE2fg=/?share_link_id=511269957959)
Miro board dat is gebruikt tijdens het gebruikersonderzoek waarin inzichten te vinden zijn over de Status Badge.

[Badges: How I Used System Thinking and Research to Evolve a Small But Essential UI element ](https://medium.com/emplifi-design/badges-small-but-essential-ui-element-16699337948b)
Onderzoek naar de kleuren en het ontwerp van een Status Badge.

[Tag - GOV.UK](https://design-system.service.gov.uk/components/tag/)
Tag implementatie bij GOV.UK. Bij de [Tag Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/62) wordt meer informatie, onderzoek en aanpassingen gedeeld. Enkele interessante comments:

- De component gebruikte voorheen hoofdletters in vetgedrukte tekst voor de tags. Dit is gewijzigd omdat onderzoek heeft aangetoond dat hoofdletters moeilijker te lezen kunnen zijn, met name voor langere tagtekst.
- De component gebruikte voorheen witte tekst op een donkere achtergrond. Onderzoek van meerdere teams wees uit dat sommige gebruikers deze zagen als knoppen en probeerden erop te klikken. Het ontwerp is gewijzigd om dit te voorkomen door een lichtere achtergrondkleur en donkerdere tekst te gebruiken.

**Comment by Rozerinay** [comment]
## Amsterdam

[Bekijk ook de toevoeging van Amsterdam in de Data Badge Discussion](https://github.com/orgs/nl-design-system/discussions/307#discussioncomment-10635116)

**Comment by Rozerinay** [comment]
## DSO Toolkit

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/signals/badge)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/docs/html-css-badge--docs)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/badge)

---

### Screenshots / Varianten

**Storybook**

Attention
<img width="88" alt="image" src="https://github.com/user-attachments/assets/4686ae73-7fe8-4fed-9719-5dc91a342c68">

Danger
<img width="77" alt="image" src="https://github.com/user-attachments/assets/6b434a47-b1c4-4a0f-bc7b-62c40c55d3d4">

Error
<img width="63" alt="image" src="https://github.com/user-attachments/assets/6b1f444b-5ce2-48e6-8e84-7c4296253187">

Info
<img width="60" alt="image" src="https://github.com/user-attachments/assets/4eb0ae60-c17d-4c40-aa5a-66b9add8c2a7">

Outline
<img width="76" alt="image" src="https://github.com/user-attachments/assets/44e0374f-70ed-4774-8766-de802b7ae3ea">

Plain
<img width="64" alt="image" src="https://github.com/user-attachments/assets/3c65a222-6284-4b29-8585-31025299adf1">

Primary
<img width="76" alt="image" src="https://github.com/user-attachments/assets/574e47a0-13e6-4cfc-83f6-a098637359da">

Success
<img width="76" alt="image" src="https://github.com/user-attachments/assets/c3f11116-7199-4fe8-8403-ae11678c8ddf">

Warning
<img width="83" alt="image" src="https://github.com/user-attachments/assets/41cc06ba-a880-4b0c-8a98-f3c8d13e0f5e">

---

### Notities

- Onder de naam: Badge.
- Beschikbaar in 9 statussen. 
- Iedere status heeft een unieke kleur.
- Bestaat uit een message.
- Richtlijnen beschikbaar (documentatiewebsite).

**Comment by Rozerinay** [comment]
## DUO

[Bekijk ook de toevoeging van DUO in de Data Badge Discussion](https://github.com/orgs/nl-design-system/discussions/307#discussioncomment-10637540)

**Comment by Rozerinay** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-status-badge--docs)
- [Figma](https://www.figma.com/design/iWuQsxelJzXVfCge87Og81/RODS-component?node-id=1-5876&node-type=frame&t=8xGABoxjj16w3j4V-11)
- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/packages/components-css/badge)

---

### Screenshots / Varianten

**Storybook**

Light blue
<img width="60" alt="image" src="https://github.com/user-attachments/assets/8d878c20-9467-4795-bd7b-4921decfd74f">

Blue
<img width="60" alt="image" src="https://github.com/user-attachments/assets/2177c45f-1cdb-4d7e-8885-143b8e5bdcf6">

Dark blue
<img width="60" alt="image" src="https://github.com/user-attachments/assets/376793b3-6522-45c4-b39e-8525e9ebf4e4">

Green blue
<img width="60" alt="image" src="https://github.com/user-attachments/assets/11828951-4dbe-4006-8005-5129dd150e89">

Red
<img width="60" alt="image" src="https://github.com/user-attachments/assets/c84769ff-9f91-49ea-b124-f9cb517b9010">

Dark yellow
<img width="60" alt="image" src="https://github.com/user-attachments/assets/376d27a4-2bb5-4a0a-b20b-048228e8f0ed">

**Figma**

<img width="122" alt="image" src="https://github.com/user-attachments/assets/ef6244b1-9378-4424-b5c1-fca62012ad01">

---

### Notities

- Onder de naam: Status Badge.
- Is onderdeel van de component Badge, net zoals Data Badge.
- Beschikbaar in 6 statussen.
- Iedere status heeft een unieke kleur. De statussen hebben toevallig een kleur als naam, zoals: 'Light blue'.
- Heeft in code de property status (`string`) én appearance (`string`), waarbij je bij de appearance kan kiezen uit een badge.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-status-badge--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=1354-6672&node-type=canvas&t=KAJQxeWqpIHYOHnJ-11)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/badge-status)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="107" alt="image" src="https://github.com/user-attachments/assets/2d3e8793-43e0-4251-99ca-b1721fb8276d">

Neutral
<img width="100" alt="image" src="https://github.com/user-attachments/assets/453eac49-94aa-4b92-b0fb-0d192f73edbe">

Success
<img width="102" alt="image" src="https://github.com/user-attachments/assets/5667bdfa-e9a5-4b6c-b64f-c10a71f704ac">

Warning
<img width="108" alt="image" src="https://github.com/user-attachments/assets/8b5f2498-2b62-44a7-ad8a-bd19ea2fe0c7">

Error
<img width="82" alt="image" src="https://github.com/user-attachments/assets/18517a29-364f-4292-8dae-fc246bb38070">

Safe
<img width="71" alt="image" src="https://github.com/user-attachments/assets/35132e37-9510-4614-937d-6a4f97bbe4f6">

Danger
<img width="93" alt="image" src="https://github.com/user-attachments/assets/fed33015-9dbd-4a4e-965e-f3119c338e87">

Active
<img width="86" alt="image" src="https://github.com/user-attachments/assets/824bf813-d5f1-4a45-adf3-dfcdc64c0fc9">

Inactive
<img width="104" alt="image" src="https://github.com/user-attachments/assets/990c1569-b379-4436-b585-c6f7f739731d">

Valid
<img width="79" alt="image" src="https://github.com/user-attachments/assets/ae8b429e-f480-493e-9b0c-252564ad155d">

Invalid
<img width="92" alt="image" src="https://github.com/user-attachments/assets/42b49080-f2df-4b3c-b57e-c0f4b298d447">


**Figma**

Default
<img width="73" alt="image" src="https://github.com/user-attachments/assets/b7090136-8a35-404e-b5a6-fd6122deef7d">

Informative
<img width="65" alt="image" src="https://github.com/user-attachments/assets/594de6cb-dc7a-4416-8abf-9cda7e52ad2a">

Negative
<img width="66" alt="image" src="https://github.com/user-attachments/assets/53e56c31-de42-4049-a96b-395e62f9a03c">

Positive
<img width="64" alt="image" src="https://github.com/user-attachments/assets/7eabbb67-2eb7-4459-9805-8e1bc17b9fe9">

Warning
<img width="65" alt="image" src="https://github.com/user-attachments/assets/1df8bc36-ae07-439a-91f5-e9ca2b2b71a7">

---

### Notities

- Onder de naam: Status Badge.
- Beschikbaar in 11 statussen in Storybook en 5 statussen in Figma.
- Statussen in Storybook en in Figma verschillen van elkaar.
- De statussen hebben verschillende kleuren, waarvan sommigen dezelfde kleur hebben, zoals 'safe' en 'active'.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Status Badge op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=status+badge)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Status Badge op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=Status+Badge)
